src/content/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/instrument-your-own.mdx

```diff
@@ -0,0 +1,274 @@
+        * Python and Ruby: `newrelic_lambda_wrapper.handler` (underscores)
+        * Node: `newrelic-lambda-wrapper.handler` (hyphens)
+        * For .Net you don't have to set the handler. 
+
+Note that for Go, you must make source code changes to your Lambda function to instrument it.
+Configuration changes are not enough.
+</Collapser>
+</CollapserGroup>
+
+## Shipping methods
+
+<CollapserGroup>
+<Collapser
+    id="extension"
+    title="Instrument with the extension only"
+>
+
+## Instrument with the extension only [#extension-only]
+Every agent layer also contains an Extension for sending logs and payloads to New Relic, bypassing CloudWatch. This can help reduce latency and costs as compared to CloudWatch. However, it comes at the risk of performance and reliability impacts to your function.
+
+After adding a New Relic Lambda layer, the extension will be enabled and have log shipping disabled by default.
+
+If you want to see function logs in New Relic, you’ll need to enable extension sending of function logs with an environment variable.
+
+It’s important to note that function logs are just the logs recorded by the function during its invocation.
+The extension will not send Lambda runtime logs like START, END, and REPORT lines.
+
+Extension environment variables
+</Collapser>
+
+ <Collapser
+    id="terraform"
+    title="Instrument with CloudWatch only"
+  >
+### Instrument with CloudWatch only [#cloudwatch-telemetry]
+
+As [mentioned previously](/docs/serverless-function-monitoring/aws-lambda-monitoring/get-started/architecture#cloudwatch)
+we used to recommend sending your telemetry through CloudWatch Logs. 
+
+1. Disable the extension by adding the `NEW_RELIC_LAMBDA_EXTENSION_ENABLED` environment variable to your function,
+   with the value `false`.
+2. Create a CloudWatch Logs subscription filter, to invoke the `newrelic-log-ingestion` function with the logs for your
+   function.
+   * The CLI can do this for you: `newrelic-lambda subscriptions install --function FUNCTION_NAME`
+   * Alternatively, use the AWS Console to create a subscription filter from your function's CloudWatch Log Group to
+     invoke the `newrelic-log-ingestion` lambda function. See below.
+
+## Manual process: Stream CloudWatch logs to New Relic Lambda
+
+    1. Open CloudWatch and select <DNT>**Logs**</DNT> in the left-hand menu, and then select the log group for the function you are monitoring.
+    2. Select <DNT>**Actions > Subscription filters > Create Lambda subscription filter**</DNT>.
+    3. Under <DNT>**Lambda function**</DNT>, select the `newrelic-log-ingestion` function.
+    4. Set the <DNT>**Log format**</DNT> to `JSON`.
+    5. Set the <DNT>**Subscription filter pattern to**</DNT> `?REPORT ?NR_LAMBDA_MONITORING ?"Task timed out" ?RequestId`. Alternatively, if you are using the `LOGGING_ENABLED` environment variable [stream all your logs](#stream-all-logs) to our Logs, leave this field blank.
```

Comment:
These anchor links don't work: `#stream-all-logs` and `#stream-logs`

Reply:
Fixed!